reducer/src/main/java/com/graphicsfuzz/reducer/glslreducers/EliminateInjectionMacrosVisitor.java

```diff
@@ -16,128 +16,84 @@
 import com.graphicsfuzz.reducer.reductionopportunities.MacroNames;
 
 public class EliminateInjectionMacrosVisitor extends StandardVisitor {
 
   @Override
-  public void visitBinaryExpr(BinaryExpr binaryExpr) {
-    super.visitBinaryExpr(binaryExpr);
-    cleanUpMacros(binaryExpr);
-  }
-
-  @Override
-  public void visitUnaryExpr(UnaryExpr unaryExpr) {
-    super.visitUnaryExpr(unaryExpr);
-    cleanUpMacros(unaryExpr);
-  }
-
-  @Override
-  public void visitParenExpr(ParenExpr parenExpr) {
-    super.visitParenExpr(parenExpr);
-    cleanUpMacros(parenExpr);
-  }
-
-  @Override
-  public void visitFunctionCallExpr(FunctionCallExpr functionCallExpr) {
-    super.visitFunctionCallExpr(functionCallExpr);
-    cleanUpMacros(functionCallExpr);
-  }
-
-  public void visitArrayIndexExpr(ArrayIndexExpr arrayIndexExpr) {
-    super.visitArrayIndexExpr(arrayIndexExpr);
-    cleanUpMacros(arrayIndexExpr);
-  }
-
-  public void visitMemberLookupExpr(MemberLookupExpr memberLookupExpr) {
-    super.visitMemberLookupExpr(memberLookupExpr);
-    cleanUpMacros(memberLookupExpr);
-  }
-
-  public void visitTernaryExpr(TernaryExpr ternaryExpr) {
-    super.visitTernaryExpr(ternaryExpr);
-    cleanUpMacros(ternaryExpr);
-  }
-
-  @Override
-  public void visitTypeConstructorExpr(TypeConstructorExpr typeConstructorExpr) {
-    super.visitTypeConstructorExpr(typeConstructorExpr);
-    cleanUpMacros(typeConstructorExpr);
-  }
-
-  @Override
-  public void visitIfStmt(IfStmt ifStmt) {
-    super.visitIfStmt(ifStmt);
-    if (MacroNames.isDeadByConstruction(ifStmt.getCondition())
-        || MacroNames.isIfWrapperFalse(ifStmt.getCondition())
-        || MacroNames.isIfWrapperTrue(ifStmt.getCondition())) {
-      ifStmt.setCondition(ifStmt.getCondition().getChild(0));
+  protected void visitChildFromParent(IAstNode child, IAstNode parent) {
+    super.visitChildFromParent(child, parent);
+    if (child instanceof FunctionCallExpr) {
+      Expr expr = (Expr) child;
+      if (MacroNames.isIdentity(expr)
+          || MacroNames.isZero(expr)
+          || MacroNames.isOne(expr)
+          || MacroNames.isFalse(expr)
+          || MacroNames.isTrue(expr)) {
+        parent.replaceChild(child,
+            addParenthesesIfNecessary(parent, ((FunctionCallExpr) child).getChild(1)));
+      } else if (MacroNames.isFuzzed(expr)
+          || MacroNames.isDeadByConstruction(expr)
+          || MacroNames.isSwitch(expr)
+          || MacroNames.isLoopWrapper(expr)
+          || MacroNames.isIfWrapperFalse(expr)
+          || MacroNames.isIfWrapperTrue(expr)
+      ) {
+        parent.replaceChild(child,
+            addParenthesesIfNecessary(parent, ((FunctionCallExpr) child).getChild(0)));
+      }
     }
   }
 
-  private void cleanupLoop(LoopStmt loopStmt) {
-    if (MacroNames.isLoopWrapper(loopStmt.getCondition())) {
-      loopStmt.setCondition(loopStmt.getCondition().getChild(0));
+  private IAstNode addParenthesesIfNecessary(IAstNode parent, Expr child) {
```

Comment:
This method is so short that I don't think it makes sense to have a separate method, "ifParenthesesNecessary".  Can you just inline the logic for "ifParenthesesNecessary" into this method?

Reply:
Fixed